### PR TITLE
API: de-dent docstrings before processing them with `@subsdoc`

### PR DESCRIPTION
**Pull request by j-ittner**

### CODE REVIEW

File: src/pytools/api/_decorators.py [hunk 1]
```diff
@@ -99,6 +100,11 @@ def subsdoc(
     Matches the given pattern in the docstring, and substitutes it with the given
     replacement string (analogous to :func:`re.sub`).
 
+    Prior to matching, the docstring is *de-dented*, i.e. the indentation of the
+    first line is removed from all lines. This ensures that docstrings that are
```
Comment (mtsokol) [inline src/pytools/api/_decorators.py:104]: Just a nitpick: how about rewording `the indentation of the
    first line is removed from all lines` to `the indentation is removed from all lines`?

**Review by mtsokol — APPROVED** [review]

File: src/pytools/api/_decorators.py (hunk 1 above)
Comment (j-ittner) [reply src/pytools/api/_decorators.py:104]: it's a subtle difference, but the first line is used to determine the overall indentation level, and then the indentation of all other lines is only removed up to the indentation of the first line, thus preserving indentation relative to the first line

Comment (mtsokol) [reply src/pytools/api/_decorators.py:104]: Ah! I missed that - now it's clear!
